src/system/SystemConfig.h

```diff
@@ -609,4 +609,48 @@ struct LwIPEvent;
 #define CHIP_SYSTEM_CONFIG_VALID_REAL_TIME_THRESHOLD 946684800
 #endif // CHIP_SYSTEM_CONFIG_VALID_REAL_TIME_THRESHOLD
 
+/**
+ *  @def CHIP_SYSTEM_CONFIG_USE_ZEPHYR_SOCKET_HACKS
+ *
+ *  @brief
+ *      Include missing functions for Zephyr sockets
+ *
+ *  Zephyr socket API lacks some of the functions required by CHIP, e.g. getsockname, recvmsg.
+ *
+ *  By default, provide the necessary functions on Zephyr platforms using sockets
+ */
+#ifndef CHIP_SYSTEM_CONFIG_USE_ZEPHYR_SOCKET_HACKS
+#if CHIP_SYSTEM_CONFIG_USE_SOCKETS && __ZEPHYR__
+#define CHIP_SYSTEM_CONFIG_USE_ZEPHYR_SOCKET_HACKS 1
```

Comment:
this characterization probably shouldn't be here

Reply:
Ok, could you suggest an alternative and explain why?